### ISSUE TITLE
Error on delete a container

### ISSUE THREAD

**Issue by Mte90** [JanitorTechnology/janitor#42]
I have an old container (10 months ago) of Firefox, when I click on VNC i get a blank page so I wanted to delete.
On delete I get an alert with:
```
Error: (HTTP code 404) no such container - No such container: 293f90b839179175b8406684c2149992635cbfe86e9ef39dceea732d8985ece3 
```
In any case I don't find a way to create a new one for firefox.

**Comment by jankeromnes** [comment]
Hi Daniele, thanks a lot for reporting this problem! Normally this shouldn't happen, but the symptoms you're seeing mean that Janitor's metadata point to a container ID you own that can no longer be found. Did you have anything valuable on it that we should try to recover for you?

Here is what we'll do to address this issue:
- [ ] We'll have a look at our system logs to investigate what happened to this particular container
- [ ] If we find problems that can lead to data loss, we'll try to improve our system so that these problems don't happen again
- [ ] We should try to keep Janitor's metadata up to date by using Docker to verify that what we think about its containers and images is still true before we access them, and better report and gracefully handle any differences we observe

**Comment by Mte90** [comment]
No was a demo for learning purpose so you can erase without any troubles :-)

**Comment by jankeromnes** [comment]
Also, of course:

- [ ] We'll manually fix your metadata so that you can create new Firefox containers again

**Comment by jankeromnes** [comment]
Is tomorrow morning ok for the metadata fix-up? I don't currently have access to a laptop and this manual fix requires a server restart. I can also try to rush it if you're in a hurry :)

**Comment by Mte90** [comment]
no worry the important the the bug get fixed :-D

**Comment by jankeromnes** [comment]
Ok, so your container `293f90b8` still lives on server IRILL, but somehow Janitor thought that it already lived on the new server MOZ1 (but it doesn't actually exist there, hence the errors). I will now investigate our spawning logic, and I would instinctively blame https://github.com/jankeromnes/janitor/blob/master/lib/machines.js#L345, which might force an existing container's host to be the project's new host (Firefox recently switched to moz1.janitor.technology, and so maybe your container was wrongly forced to the new host).

**Comment by jankeromnes** [comment]
TL;DR: Should be fixed by https://github.com/janitortechnology/janitor/pull/43.

The "host resetting logic" assumed that you can create multiple machines for a project, and thus that `machine` in `getOrCreateMachine()` is always "new", so forcing `machine.docker.host` to be the Docker host where `spawn()` would actually create the container made sense.

What happened instead, causing this bug, is that `getOrCreateMachine()` was returning your existing machine, but still forcing its `machine.docker.host` to the project's new Docker host (as mentioned earlier, Firefox recently moved from "janitor.technology" to "moz1.janitor.technology").

Until the PR is merged and deployed, the same problem will happen to anyone with contributions of Firefox, Servo or Thunderbird older than Friday, 24 Feb 2017 (currently, older than 10 days), so once the PR is deployed we should:

- [ ] manually fix the metadata of any other container affected by the same bug, restoring `machine.docker.host` to `'janitor.technology'`

**Comment by jankeromnes** [comment]
We just deployed pull request #43, meaning that this immediate problem should not happen again in production going forward. (But I'm leaving the higher-level "we should use Docker more to keep machine meta-data up-to-date" item unchecked.)

I also had a quick look through the database to look for machines that were affected, by looking for containers that are both:

- Currently running on `janitor.technology`
- Have `machine.docker.host` wrongly set to `moz1.janitor.technology`

I found 3 containers, including yours @Mte90, and fixed their `machine.docker.host` back to `janitor.technology`. You should now be able to delete it without issues, could you please confirm that it works? Thanks again! :)

**Comment by Mte90** [comment]
yep delete successfully!

**Comment by jankeromnes** [comment]
Woohoo! Thanks a lot! 🎉😄🎉

**Comment by jankeromnes** [comment]
It appears that a 4th container, created 9 months ago, was also affected by this. I've manually fixed its metadata like for the other 3 affected containers.
